echo-notifications/src/main/java/com/netflix/spinnaker/echo/notification/InteractiveNotificationCallbackHandler.java

```diff
@@ -0,0 +1,150 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import retrofit.Endpoints;
+import retrofit.RestAdapter;
+import retrofit.client.Response;
+import retrofit.converter.JacksonConverter;
+import retrofit.http.Body;
+import retrofit.http.Header;
+import retrofit.http.POST;
+
+@Component
+public class InteractiveNotificationCallbackHandler {
+  private final Logger log = LoggerFactory.getLogger(InteractiveNotificationCallbackHandler.class);
+
+  private Ok3Client spinnakerServiceClient;
+  private List<InteractiveNotificationService> notificationServices;
+  private Environment environment;
+  private Map<String, SpinnakerService> spinnakerServices = new HashMap<>();
+
+  @Autowired
+  public InteractiveNotificationCallbackHandler(
+      Ok3Client spinnakerServiceClient,
+      List<InteractiveNotificationService> notificationServices,
+      Environment environment) {
+    this.spinnakerServiceClient = spinnakerServiceClient;
+    this.notificationServices = notificationServices;
+    this.environment = environment;
+  }
+
+  public InteractiveNotificationCallbackHandler(
+      Ok3Client spinnakerServiceClient,
+      List<InteractiveNotificationService> notificationServices,
+      Environment environment,
+      Map<String, SpinnakerService> spinnakerServices) {
+    this(spinnakerServiceClient, notificationServices, environment);
+    this.spinnakerServices = spinnakerServices;
+  }
+
+  /**
+   * Processes a callback request from the notification service by relaying it to the downstream
+   * Spinnaker service that originated the message referenced in the payload.
+   *
+   * @param source The unique name of the source of the callback (e.g. "slack")
+   * @param request The request received from the notification service
+   */
+  public ResponseEntity<String> processCallback(
+      final String source, RequestEntity<String> request) {
+    log.debug("Received interactive notification callback request from " + source);
+
+    InteractiveNotificationService notificationService = getNotificationService(source);
+
+    if (notificationService == null) {
+      throw new NotFoundException("NotificationService for " + source + " not registered");
+    }
+
+    final Notification.InteractiveActionCallback callback =
+        notificationService.parseInteractionCallback(request);
+
+    SpinnakerService spinnakerService = getSpinnakerService(callback.getServiceId());
+
+    log.debug("Routing notification callback to originating service " + callback.getServiceId());
+
+    // TODO: error handling (retries?)
```

Comment:
You should address this todo or remove the commented out code

Reply:
I'm currently unable to respond to messages in the chat to create a thread, pending help from Slack support, which is why I got stuck here and added the `TODO`. Was hoping to tackle that in a later PR if that's OK. The next best thing would be to replace the buttons in the original message with some error information, which sounded less than ideal from a UX perspective. @erikmunson @gcomstock Any thoughts on this?

Reply:
Feel free to keep the `todo`, I'd just remove the commented out retry code.

Reply:
Thanks for calling me out Luis and sorry for my delay. Sounds like you took the best course of action for now. Enjoy your trip!